src/core/services/global-context-service.ts

```diff
@@ -0,0 +1,70 @@
+import { utils } from '../index';
+
+export enum GlobalContextCategories {
+	site = 'site',
+	page = 'page',
+	tracking = 'tracking',
+	partners = 'partners',
+}
+
+export class GlobalContextService {
+	private logGroup = 'GlobalContextService';
+
+	constructor() {
+		if (!window.fandomContext) {
+			this.createEmptyContext();
+		}
+	}
+
+	private createEmptyContext(): void {
+		utils.logger(this.logGroup, 'Creating empty fandomContext');
+		window.fandomContext = {
+			site: {} as ISiteParameters,
+			page: {} as IPageParameters,
+			tracking: {} as ITrackingParameters,
```

Comment:
```suggestion
			tracking: {},
```
should work as tracking is `Partial`, same for `partners` (and could be same for site and page as they should be `Partial` too imho)